cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -252,25 +235,31 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
             vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<ReconciliationState>executeBlocking(
                 future -> {
                     try {
-                        String clusterCaName = AbstractModel.getClusterCaName(reconciliation.name());
+                        String clusterCaCertName = AbstractModel.getClusterCaName(name);
+                        String clusterCaKeyName = AbstractModel.getClusterCaKeyName(name);
+                        String clientsCaCertName = KafkaCluster.clientsPublicKeyName(name);
+                        String clientsCaKeyName = KafkaCluster.clientsCASecretName(name);
+                        Secret clusterCaCertSecret = null;
+                        Secret clusterCaKeySecret = null;
+                        Secret clientCaCertSecret = null;
+                        Secret clientCaKeySecret = null;
                         List<Secret> clusterSecrets = secretOperations.list(reconciliation.namespace(), caLabels);
-                        Secret clusterCa = findSecretWithName(clusterSecrets, clusterCaName);
-                        SecretCertProvider secretCertProvider = new SecretCertProvider();
-                        Secret secret;
-                        X509Certificate currentCert = cert(clusterCa, CLUSTER_CA_CRT);
-                        boolean needsRenewal = currentCert != null && certNeedsRenewal(tlsCertificates, currentCert);
-                        final Map<String, String> newData;
-                        if (tlsCertificates != null && !tlsCertificates.isGenerateCertificateAuthority()) {
-                            newData = checkProvidedCert(reconciliation, clusterCaName, clusterCa, currentCert, needsRenewal);
-                        } else {
-                            if (clusterCa == null || needsRenewal) {
-                                newData = createOrRenewCert(reconciliation, tlsCertificates, clusterCaName, clusterCa, currentCert);
-                            } else {
-                                log.debug("{}: The cluster CA {} already exists and does not need renewing", reconciliation, clusterCaName);
-                                newData = clusterCa.getData();
+                        for (Secret secret : clusterSecrets) {
```

Comment:
This seems to be the same as the `sort` method. Is there no way how to do this only once in one place? It seems we would need to change the same thing in multiple places.

Reply:
It's finding different secrets by name and initialising variables/fields with them, but they're different secrets and different fields. We could, I suppose, decode them (once) into a `Map` keyed by name, and then use `get` to initialize the things, but I don't think that would be so much better. Or we could use `findSecretByName()`, which I prefer, I guess. I mean it's inefficient, but this is hardly going to be hot code.